concept_mapper_experiment/01_create_concept_to_id_mapper.py

```diff
@@ -0,0 +1,231 @@
+            line = line.strip()
+            match_obj = re.search("MH = ", line)
+
+            if match_obj is not None:
+                concept = re.sub("MH = ", "", line).lower()
+
+            match_obj = re.search("UI = ", line)
+            if match_obj is not None:
+                concept_id = re.sub("UI = ", "", line).lower()
+                data_rows.append({"mesh_id": concept_id, "mesh_heading": concept})
+
+        for line in tqdm.tqdm(chem_infile):
+            line = line.strip()
+            match_obj = re.search("MH = ", line)
+
+            if match_obj is not None:
+                concept = re.sub("MH = ", "", line).lower()
+
+            match_obj = re.search("UI = ", line)
+            if match_obj is not None:
+                concept_id = re.sub("UI = ", "", line).lower()
+                data_rows.append({"mesh_id": concept_id, "mesh_heading": concept})
+
+    concept_df = pd.DataFrame.from_records(data_rows)
+    concept_df >> ply.call("to_csv", str(mesh_map_file), sep="\t", index=False)
+else:
+    concept_df = pd.read_csv(str(mesh_map_file), sep="\t")
+print(concept_df.shape)
+concept_df >> ply.slice_rows(10)
+
+# # Extract Species IDs
+
+species_id_url = "https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump.tar.gz"
+
+species_file = f"output/intermediate_files/{Path(species_id_url).name}"
+if not Path(species_file).exists():
+    os.system(f"wget {species_id_url} -O {species_file}")
+    Path("output/intermediate_files/species_temp").mkdir(exist_ok=True, parents=True)
+    os.system(
+        f"gunzip -c {species_file} | tar xf - -C output/intermediate_files/species_temp"
+    )
+
+# +
+species_map_file = Path("output/species_id_map.tsv")
+if not species_map_file.exists():
+    with open(
+        "output/intermediate_files/species_temp/names.dmp", "r"
+    ) as species_infile:
+        data_rows = []
+        for idx, line in enumerate(species_infile):
+            fields = re.sub("\t", "", line.strip()).split("|")
+
+            if fields[3] == "scientific name":
+                data_rows.append(
+                    {"species_id": int(fields[0]), "species_name": fields[1].lower()}
+                )
+
+        species_df = pd.DataFrame.from_records(data_rows)
+        species_df >> ply.call("to_csv", str(species_map_file), sep="\t", index=False)
+else:
+    species_df = pd.read_csv(str(species_map_file), sep="\t")
+
+print(species_df.shape)
+species_df >> ply.slice_rows(10)
+# -
+
+# # Extract Gene IDS
+
+gene_id_url = "https://ftp.ncbi.nlm.nih.gov/gene/DATA/GENE_INFO/All_Data.gene_info.gz"
+
+gene_file = Path(f"output/intermediate_files/{Path(gene_id_url).name}")
+if not gene_file.exists():
+    os.system(f"wget {gene_id_url} -O {gene_file}")
+
+gene_map_file = Path("output/gene_id_map.tsv")
+if not gene_map_file.exists():
+    with gzip.open(str(gene_file), "rt") as infile:
+        data_rows = []
+        for idx, line in tqdm.tqdm(enumerate(infile)):
+            if idx == 0:
```

Comment:
Are you skipping a header here? Might be worth a short comment saying you don't need the header, just for future reference.